user-management/src/main/scala/com/normation/plugins/usermanagement/api/UserManagementApi.scala

```diff
@@ -185,24 +217,53 @@ class UserManagementApiImpl(
     val schema = UserManagementApi.GetUserInfo
     def process0(version: ApiVersion, path: ApiPath, req: Req, params: DefaultParams, authzToken: AuthzToken): LiftResponse = {
 
-      // This is just a compat hub done so that we can see all users. There will be problems
       (for {
-        users    <- userRepo.getAll()
-        allRoles <- RudderRoles.getAllRoles
+        users     <- userRepo.getAll()
+        allRoles  <- RudderRoles.getAllRoles
+        file       = userService.authConfig
+        roles      = allRoles.values.toSet
+        jsonUsers <-
+          ZIO
+            .foreach(users)(u => {
+              implicit val currentRoles: Set[Role] = roles
+              userRepo.getLastPreviousLogin(u.id).map { lastSession =>
+                file.users.get(u.id) match {
+                  case None    => transformProvidedUser(u, lastSession)
+                  case Some(x) => {
+                    // we need to update the status to the latest one from database
+                    val currentUserDetails = x.copy(status = u.status)
+
+                    // since file definition does not depend on session use the file user as base for file-managed users
+                    val fileProviderInfo = JsonProviderInfo.fromUser(x, DefaultAuthBackendProvider.FILE)
+
+                    if (u.managedBy == DefaultAuthBackendProvider.FILE) {
+                      transformUser(
+                        currentUserDetails,
+                        u,
+                        Map(fileProviderInfo.provider -> fileProviderInfo),
+                        lastSession.map(_.creationDate)
+                      ).withRoleCoverage(currentUserDetails)
+                    } else {
+
+                      // we need to merge the two users, the one from the file and the one from the session
+                      // depending on provider property configuration, we should merge or override roles, by default merge
+                      val providerProperties = authProvider.getProviderProperties()
+                      val overrideRoles      = providerProperties.get(u.managedBy).map(_.overridesRoles).getOrElse(false)
+                      // merge at the level of JsonUser because last session roles are just String
+                      val providedUser       = transformProvidedUser(u, lastSession)
+                      if (overrideRoles) {
+                        providedUser.addProviderInfo(fileProviderInfo)
+                      } else {
+                        providedUser.merge(fileProviderInfo).withRoleCoverage(x)
+                      }
```

Comment:
There is something strange with that API endpoint: I try to log/unlog/relog, and still get "never log in"
Plus I thing we want to adapt text to "first log-in", else it's a bit confusing

![image](https://github.com/Normation/rudder-plugins/assets/44649/5ceb097c-9612-44c9-b115-0fdb3002fe89)

Reply:
AH ! Perhaps I have an idea:
- the first log-in doesn't have an previous sessions
- there seems to be a bug when the user is auto logged-out because sessions expired, the base is not correctly filled - so still no closed sessions
- and finally when I logged out before sessions expired and logged in again, I got the correct behavior 

![image](https://github.com/Normation/rudder-plugins/assets/44649/3ed80c76-2975-4d37-a3d9-06a849053d4e)

Reply:
It's because the query for the previous login adds a `where enddate is not null`, maybe we should parameterize that predicate ? 
It's useful [here](https://github.com/Normation/rudder/blob/backports/7.3.12/24146/webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/snippet/UserInformation.scala#L77) where we want the current user to know the last time they logged in.
But in user-management we want the 'current' session to be considered